src/actions/measurements.js

```diff
@@ -263,34 +264,45 @@ export const parseMeasurementsJSON = (json) => {
     );
   });
 
-  return {collections, defaultCollection: json["default_collection"]};
-};
-
-export const loadMeasurements = ({collections, defaultCollection}) => (dispatch, getState) => {
-  const { tree, controls } = getState();
-  if (!tree.loaded) {
-    throw new Error("tree not loaded");
-  }
-
   const collectionKeys = collections.map((collection) => collection.key);
-  let defaultCollectionKey = defaultCollection;
+  let defaultCollectionKey = json["default_collection"];
   if (!collectionKeys.includes(defaultCollectionKey)) {
     defaultCollectionKey = collectionKeys[0];
   }
-
-  // Get the collection to display to set up default controls
-  const collectionToDisplay = getCollectionToDisplay(collections, controls.measurementsCollectionKey, defaultCollectionKey);
-  const newControls = getCollectionDisplayControls(controls, collectionToDisplay);
-  const queryParams = createMeasurementsQueryFromControls(newControls, collectionToDisplay, defaultCollectionKey);
-
-  dispatch({
-    type: LOAD_MEASUREMENTS,
+  const collectionToDisplay = collections.filter((collection) => collection.key === defaultCollectionKey)[0];
+  return {
+    loaded: true,
+    error: undefined,
     defaultCollectionKey,
     collections,
-    collectionToDisplay,
-    controls: newControls,
-    queryParams
-  });
+    collectionToDisplay
+  }
+};
+
+export const loadMeasurements = (measurementsData, dispatch) => {
+  let measurementState = getDefaultMeasurementsState();
+  let warningMessage = "";
+  if (measurementsData === undefined) {
+    // eslint-disable-next-line no-console
+    console.debug("No measurements JSON fetched");
```

Comment:
For a dataset without measurements `measurementsData=false` so we're getting the warning message when we shouldn't. This happens because while the dispatch object sets `measurementsData: undefined` that won't override the default arg value of `createStateFromQueryOrJSONs({measurementsData = false, ...})`. This is a feature / bug of JS depending on your point of view:

```js
function test({aaa=false}={}) {console.log("inside", aaa)}
test() // inside false
test({aaa: undefined}) // inside false
```

You can use `null` if you want, but I'd suggest `false`

Reply:
Ah good catch, I'll go with `false`.